frontend/app/mocks/user-api.server.ts

```diff
@@ -0,0 +1,51 @@
+import { HttpResponse, http } from 'msw';
+import { z } from 'zod';
+
+import { db } from './db';
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('user-api.server');
+
+const { CDCP_API_BASE_URI } = getEnv();
+
+const userSubmittedApiSchema = z.object({
+  id: z.string().optional(),
+  email: z.string().optional(),
+  emailVerified: z.boolean().optional(),
+  userAttributes: z.array(
+    z.object({
+      name: z.string(),
+      value: z.string(),
+    }),
+  ),
+});
+
+export function getUserApiMockHandlers() {
+  log.info('Initializing User API mock handlers');
+
+  return [
+    http.post(`${CDCP_API_BASE_URI}/api/v1/users`, async ({ params, request }) => {
```

Comment:
If `CDCP_API_BASE_URI` is configured to be the actual API, requests will still hit this mock.

```suggestion
    http.post('https://api.cdcp.example.com/api/v1/users', async ({ params, request }) => {
```

We should also add the ability to enable/disable this API mock. See:
- https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/mocks/node.ts
- https://github.com/DTS-STN/canadian-dental-care-plan/blob/0f4bb6dde6505c9c66ed00164525c21a6c78540d/frontend/.env.example#L203